compiler/luci/pass/src/QuantizeWeightsOnly.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "QuantizeWeightsOnly.h"
+#include "QuantizationUtils.h"
+
+#include <luci/Service/Nodes/CircleConst.h>
+#include <luci/Log.h>
+
+#include <cmath>
+#include <vector>
+#include <functional>
+#include <limits>
+
+using namespace luci;
+
+namespace
+{
+
+using IterFunc = std::function<void(uint32_t *, loco::TensorShape &, int32_t)>;
+
+void iterate_per_channel(CircleConst *node, int32_t &channel_dim_index, IterFunc func)
```

Comment:
- `iterate_per_channel` exists in `QuantizeWeights.cpp` and `QuantizeDequantizeWeightsPass`. Both are almost same, but slightly different. 

- It may be better to move this into `QuantizationUtils`. I don't know the history for the duplication (maybe intended). I guess several... performance? (by assuming channel_dim_index is always zero). But I am not sure.

Reply:
There is no performance reason. I think the function was simply missed during the past refactoring job. It would be great if you can help :)